src/components/common/OptionPopper.tsx

```diff
@@ -65,10 +67,11 @@ const OptionPopper = ({
         <div>
           <Table className={classes.tablePopper}>
             <TableBody>
-              {options.map((option: Option) => {
+              {options.map((option: Option, index: number) => {
```

Comment:
this whole block can be simplified a tiny bit (basically removing the `return` statement) if you want to make a final commit
```
              {options.map((option: Option, index: number) => (
                <TableRow
                  hover
                  key={id + String(index)}
                  classes={{ hover: optionStyle.menuHover }}
                  onClick={option.onClick}
                >
                  <TableCell classes={{ root: option.styles }}>
                    <Typography variant="body2">{option.name}</Typography>
                  </TableCell>
                </TableRow>
              ))}
```